woocommerce/woocommerce
# Revert "Set stock quantity value to 0 by default (#48448)"

This reverts commit 0652c2eb4fa00c1f2ea990d675aee8aba069ee97.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





The PR https://github.com/woocommerce/woocommerce/pull/48448 set stock quantity to 0 by default. However, the change had bigger implications than we thought. It initially breaks some use cases for Product Bundles and potentially could break backward compatibility. This is why we decided to revert the change.

More context here (p1719342549204909-slack-C01DT6U03HC)

With the changes reverted here, you will see again:
- Set the stock quantity to 1 by default in the UI.
- It's possible to set the stock quantity to `null`.
- `get_stock_quantity` can return `null` again.

⚠️ This is not a simple revert. There are a few changes in the original PR that are necessary due to the linter. A simple revert won't be accepted because the original files need to be updated to pass the linter checks before modifying them.

You can check [this issue](https://github.com/woocommerce/woocommerce/issues/39475) if you want additional details.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch.
2. Go to Products > Add New > Inventory.
3. Check `Track stock quantity for this product`.
4. Verify that `Quantity` is 1.
5. Create a product and confirm that the stock management works well.
6. Try a few extensions that interact with the stock quantity like `WooCommerce Product Bundles` and `WooCommerce Back in Stock Notifications`. If you remember more extensions, please test them.
7. Perform some smoke tests. Verify that the Stock status works as expected and that the available stock matches what it should. Create multiple orders from the product page, and very everything works ok.
8. Enable the new product editor under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
9. Go to Products > Add New > Inventory.
10. Press `Track inventory`.
11. Verify the `Available stock` is 1.
12. Make more smoke tests.
13. Test with variable products as well.
14. Go to `/wp-admin/admin.php?page=wc-settings&tab=products&section=inventory` and disable `Manage stock`.
15. Create a few orders and verify everything works as expected.
16. Re-enable the `Manage stock` option. Everything should work well.
17. Since this affects multiple areas, please test any other use case you can think of.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
